SlackBotPrototype/API/MessageParser.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Console = System.Console;
+using edu.stanford.nlp.ling;
+using edu.stanford.nlp.tagger.maxent;
+
+namespace API
+{
+	public class MessageParser : IMessageParser
+	{
+		private readonly string _modelsDirectory;
+		
+		public MessageParser(string nlpFolder)
+		{
+			_modelsDirectory = nlpFolder + @"\models";
+		}
+
+		public string HandleRequest(string message)
+		{
+			var wordSet = BuildWordSet(message);
+			var weatherProvider = new DarkSkyWeatherProvider(ConfigConstants.DarkSkyApiSecret);
+			
+			if (wordSet.Contains("wear", StringComparer.InvariantCultureIgnoreCase) &&
+			    wordSet.Contains("I", StringComparer.InvariantCultureIgnoreCase))
+			{
+				var rawForecast = weatherProvider.GetForecastRaw(DateTime.Now);
+				var icon = rawForecast.Daily.Data.FirstOrDefault()?.Icon;
+				
+				// Skip accounting for temperature for now, could have thresholds for certain apparel
+				switch (icon)
+				{
+					case "clear-day":

```

Comment:
👍 Could use the map concept for matching up the "commands" as well, i.e. map strings to match on to function calls.